### DISCUSSION TITLE
Support Mock Matching using Post Body

### DISCUSSION THREAD

**Discussion by hecktarzuli** [mswjs/msw#1721] [Q&A] [answered]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

It's great we can mock a REST endpoint with `.post()`. The problem is, with rest APIs, different post payloads to a given url may return different responses.

It would be really nice if `rest.post()` accepted an object (or something) where we could declare separate mocks for both a url AND payload.  Currently we have to put all of the matching and splitting logic within the 1 mock handler.

**Comment by kettanaito** [comment, marked as answer]
Hi, @hecktarzuli. 

Please take a look at a similar question about matching against request query parameters: https://github.com/mswjs/msw/discussions/1315#discussioncomment-3108065. The approach to implementing the match against the request body will be quite the same (here's a more relevant comment on the topic https://github.com/mswjs/msw/discussions/1315#discussioncomment-5526210). 

MSW provides a versatile API for you to implement such a helper utility and use across your request handlers. This is not something we will ship out of the box.
